### PR TITLE
elastic storage: fixed delete endpoints, improved error handling, add…

### PR DESCRIPTION
**Pull request by JuliaFernee**
…ed usage to README, plus excluded stuff to gitignore

### CODE REVIEW

File: engine_mgo.go [hunk 1]
```diff
@@ -64,12 +64,13 @@ func (eng *mongoEngine) EnsureIndexes(collection Collection) {
 
 }
 
-func (eng *mongoEngine) Drop(collection Collection) {
+func (eng *mongoEngine) Drop(collection Collection) (bool, error){
```
Comment (ghost) [inline engine_mgo.go:67]: this now claims to return errors, but doesn't return errors in reality

File: engine_elasticsearch.go [hunk 2]
```diff
@@ -18,6 +18,7 @@ type elasticEngine struct {
 
 func NewElasticEngine(elasticURL string, indexName string) Engine {
 	transport := &http.Transport{MaxIdleConnsPerHost: 30}
+	fmt.Printf("elasticUrl = %s, indexName=%s", elasticURL, indexName)
```
Comment (ghost) [inline engine_elasticsearch.go]: remove this please.

File: engine_elasticsearch.go [hunk 3]
```diff
@@ -164,17 +175,17 @@ func (ee elasticEngine) Ids(collection Collection, stopchan chan struct{}) (chan
...
 		panic(err)
 	}
-
 	switch {
 	case res.StatusCode == 200:
 	case res.StatusCode == 400:
 		res.Body.Close()
 		return nil, ErrInvalidQuery
+	case res.StatusCode == 404:
+		return nil, ErrNotFound
```
Comment (ghost) [inline engine_elasticsearch.go:187]: If a caller gets "not found" when running a query, would they expect that to mean they their search returned nothing, or that the collection they were searching in doesn't exist?  Maybe it should be explicitly "collection not found" ?

File: app.go [hunk 4]
```diff
@@ -262,15 +260,27 @@ func (ah *apiHandlers) dropHandler(w http.ResponseWriter, r *http.Request) {
...
 	if err != nil {
+
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
-	ah.engine.Drop(coll)
+
+	ok, err := ah.engine.Drop(coll)
+
+	if err !=nil {
```
Comment (ghost) [inline app.go:270]: How can we be sure it's a bad request here?  if elastic search was unavailable, we'd get an error, but it's not a bad request.

File: README.md [hunk 5]
```diff
@@ -3,3 +3,92 @@ An experimental RESTful storage API with various pluggable backends.
...
+}]
+}
+```
+
+GET http://localhost:8080/organisations/013f7fa7-aa26-3e20-84f1-fb8e5f7383ff  
+DELETE http://localhost:8080/organisations/013f7fa7-aa26-3e20-84f1-fb8e5f7383ff  
+
+### Bulk Document endpoints usage
+PUT http://localhost:8080/people/  
+```
```
Comment (ghost) [inline README.md:44]: We should not put data in a public repository if it is derived from factset data.  I don't believe our contract allows it.

File: app.go (hunk 4 above)
Comment (JuliaFernee) [inline app.go:270]: Fixed

File: README.md (hunk 5 above)
Comment (JuliaFernee) [inline README.md:44]: Fixed

File: engine_elasticsearch.go (hunk 3 above)
Comment (JuliaFernee) [inline engine_elasticsearch.go:187]: Can we assume in this version that it's the same "collection not found" or a "record(s) in the collection not found" - both follow the same query pattern and natively ES returns 404 for either

File: engine_mgo.go (hunk 1 above)
Comment (JuliaFernee) [inline engine_mgo.go:67]: Added TODO. Don't know what kind of solution is going to be here eventually. I assune the mongo part is still in progress (saw other TODOs in the same file)

File: engine_elasticsearch.go (hunk 2 above)
Comment (JuliaFernee) [inline engine_elasticsearch.go]: done

**Pull request merged**